### DISCUSSION TITLE
How to configure a history per job ?

### DISCUSSION THREAD

**Discussion by Furvent** [jhuckaby/Cronicle#578] [Q&A] [answered]
Hello !
First of all, thank you to the creator and contributors of CRONicle for this magnificent and so practical tool.
I use it in a professional setting, and with my team we would like to know if it was possible to configure a specific history for each job.
Example :
- for job number 1, which would run once a day, I would like to keep the logs of 10 or 20 runs,
- for job number 2, which would run every minute, I would like to keep the logs of 1000 or 2000 runs.

It doesn't seem to be possible with the config file, but maybe I didn't look hard enough.
Otherwise the other possibility would be to modify the code directly, and in this case does someone who knows how the program works well have a way to implement this functionality?
Thanks for reading.

**Comment by mcfetz** [comment]
You can define a individual job log expire in every job definition. It overrides the general setting from the config file.

**Comment by jhuckaby** [comment, marked as answer]
Inside your running job, print this JSON to STDOUT:

```js
{ "log_expire_days": 5 }
```

Make sure it is all on one line, with an EOL at the end.  Remember, JSON syntax is very picky about quotes and curly braces.  Gotta get them all exactly right.

This will change the expiration days for the current running job only.

**Comment by Furvent** [comment]
Thanks a lot for your answers !
